controlplane/kubeadm/internal/etcd_client_generator.go

```diff
@@ -60,28 +57,30 @@ func (c *etcdClientGenerator) forNode(ctx context.Context, name string) (*etcd.C
 	return customClient, nil
 }
 
-// forLeader takes a list of nodes and returns a client to the leader node
-func (c *etcdClientGenerator) forLeader(ctx context.Context, nodes *corev1.NodeList) (*etcd.Client, error) {
-	var errs []error
-
-	for _, node := range nodes.Items {
-		client, err := c.forNode(ctx, node.Name)
-		if err != nil {
-			errs = append(errs, err)
-			continue
-		}
-		defer client.Close()
-		members, err := client.Members(ctx)
-		if err != nil {
-			errs = append(errs, err)
-			continue
-		}
-		for _, member := range members {
-			if member.ID == client.LeaderID {
-				return c.forNode(ctx, member.Name)
-			}
-		}
+func (c *etcdClientGenerator) forAny(ctx context.Context, nodes *corev1.NodeList) (*etcd.Client, error) {
```

Comment:
forNode() is taking node name as input, to be consistent either this can that []string or forNode() takes Node as argument.